loopy/translation_unit.py

```diff
@@ -766,19 +747,34 @@ def _collective_transform(*args, **kwargs):
 
 
 def update_table(callables_table, clbl_id, clbl):
+    """
+    Returns a tuple ``new_clbl_id, new_callables_table`` where
+    *new_callables_table* is a copy of *callables_table* with *clbl* in its
+    namespace. *clbl* is referred to in *new_callables_table*'s namespace by
+    *new_clbl_id*.
+
+    :arg clbl_id: An instance of :class:`str` or
+        :class:`~loopy.library.reduction.ReductionOpFunction` based on which
+        the unique identifier, *new_clbl_id* , is to be chosen.
+    """
```

Comment:
Shouldn't this also update the internal name then?

Reply:
Yep, d2d12ac06aa15715ad0266be895b560b50b2b589.